testing/internal/e2e/tests/database/migration_test.go

```diff
@@ -0,0 +1,289 @@
+			"token", "create",
+			"-no-default-policy=true",
+			"-policy="+boundaryPolicyName,
+			"-policy="+kvPolicyName,
+			"-orphan=true",
+			"-period=20m",
+			"-renewable=true",
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	var tokenCreateResult vault.CreateTokenResponse
+	err = json.Unmarshal(output.Stdout, &tokenCreateResult)
+	require.NoError(t, err)
+	credStoreToken := tokenCreateResult.Auth.Client_Token
+	t.Log("Created Vault Cred Store Token")
+	newVaultCredentialStoreId := boundary.CreateNewCredentialStoreVaultCli(t, ctx, newProjectId, vaultAddr, credStoreToken)
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"credential-libraries", "create", "vault",
+			"-credential-store-id", newVaultCredentialStoreId,
+			"-vault-path", c.VaultSecretPath+"/data/"+privateKeySecretName,
+			"-name", "e2e Automated Test Vault Credential Library",
+			"-credential-type", "ssh_private_key",
+			"-format", "json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	var newCredentialLibraryResult credentiallibraries.CredentialLibraryCreateResult
+	err = json.Unmarshal(output.Stdout, &newCredentialLibraryResult)
+	require.NoError(t, err)
+	newCredentialLibraryId := newCredentialLibraryResult.Item.Id
+	t.Logf("Created Credential Library: %s", newCredentialLibraryId)
+
+	// Create a session. Uses Boundary in a docker container to do the connect in order to avoid
+	// modifying the runner's /etc/hosts file. Otherwise, you would need to add a `127.0.0.1
+	// localhost boundary` entry into /etc/hosts.
+	buf = bytes.NewBuffer(nil)
+	ebuf = bytes.NewBuffer(nil)
+	_, err = b.Resource.Exec(
+		[]string{
+			"boundary", "authenticate", "password",
+			"-addr", b.UriNetwork,
+			"-auth-method-id", dbInitInfo.AuthMethod.AuthMethodId,
+			"-login-name", dbInitInfo.AuthMethod.LoginName,
+			"-password", "env://E2E_TEST_BOUNDARY_PASSWORD",
+			"-format", "json",
+		},
+		dockertest.ExecOptions{
+			StdOut: buf,
+			StdErr: ebuf,
+			Env:    []string{"E2E_TEST_BOUNDARY_PASSWORD=" + dbInitInfo.AuthMethod.Password},
+		},
+	)
+	require.NoError(t, err)
+	require.Empty(t, ebuf)
+	var authenticationResult boundary.AuthenticateCliOutput
+	err = json.Unmarshal(buf.Bytes(), &authenticationResult)
+	require.NoError(t, err)
+	auth_token := authenticationResult.Item.Attributes["token"].(string)
+
+	connectTarget := infra.ConnectToTarget(t, pool, network, b.UriNetwork, auth_token, newTargetId)
+	t.Cleanup(func() {
+		pool.Purge(connectTarget.Resource)
+	})
+	_, err = pool.Client.WaitContainer(connectTarget.Resource.Container.ID)
+	require.NoError(t, err)
+	buf = bytes.NewBuffer(nil)
+	ebuf = bytes.NewBuffer(nil)
+	err = pool.Client.Logs(docker.LogsOptions{
+		Container:    connectTarget.Resource.Container.ID,
+		OutputStream: buf,
+		ErrorStream:  ebuf,
+		Follow:       true,
+		Stdout:       true,
+		Stderr:       true,
+	})
+	require.NoError(t, err)
+	require.Empty(t, ebuf)
+	t.Log("Created session")
```

Comment:
Maybe all of this should be a helper too, since it's all still setup for the line below, where the important part is happening.

Reply:
Made an attempt to do this. The function signature is a bit long, though.